protos/configuration.proto

```diff
@@ -99,6 +99,12 @@ message RaftConfig {
   // pass between heartbeats. That is, a leader sends heartbeat
   // messages to maintain its leadership every HeartbeatTick ticks.
   uint32 heartbeat_ticks = 3;
+
+  // Limits the max number of in-flight blocks (i.e. Raft messages).
+  uint32 max_inflight_blocks = 4;
+
+  // Take a snapshot when cumulative data since last snapshot exceeds a certain size in bytes.
+  uint64 snapshot_interval_size = 5;
```

Comment:
can't we use the `types.RaftConfig` instead of `RaftConfig` in the `shared_config.go`? I remember asking this question in the past and don't remember your response.

Reply:
Actually, if I recall correctly, you asked me specifically not to use the `type.*` protobuf messages in the config package?
It used to be like that (reused protos in config) but I changed it when I did the shared config PR a while pack...
There is a duplication of  structs between `config.*` and `types.*` in many other levels as well...

Reply:
I see. Maybe the reason was to expand the `config.yml` independently of the proto message.